src/redux/data.ts

```diff
@@ -748,7 +761,12 @@ export const transactionsReceived = (
     }, 60000);
   }
   const txHashes = parsedTransactions.map(tx => ethereumUtils.getHash(tx));
-  const updatedPendingTransactions = pendingTransactions.filter(
+  const decoratedPendingTransactions = await maybeFetchF2CHashForPendingTransactions(
+    {
+      pendingTransactions,
+    }
+  );
```

Comment:
Our only source of tx data is this event handler, so when we get new data we need to check if any f2c deposits have tx hashes available yet. We add the hash to the pending transaction if it exists, and then filter these out of pending on the next line.

Reply:
Makes sense, looks clean!